frontend/src/utilities/hooks/useWidget.js

```diff
@@ -4,17 +4,28 @@ import { useAxios } from './useAxios';
 
 const sqlApi = 'https://cdb-cdn.resilienceatlas.org/user/ra/api/v2/sql';
 
-export const useWidget = ({ slug, geojson }, { analysisQuery, analysisBody }) => {
+export const useWidget = ({ slug, geojson }, { type, analysisQuery, analysisBody }) => {
+  const isCOGLayer = useMemo(() => type === 'cog', [type]);
   const query = useMemo(() => {
     if (analysisBody) {
-      const { assetId } = JSON.parse(analysisBody);
+      const { assetId, params } = JSON.parse(analysisBody);
+      let parsedQuery = analysisQuery;
+
+      if (isCOGLayer && params && typeof params === 'object') {
+        params &&
```

Comment:
I think this line is not needed since you're already checking if `params` exists above.